packages/features/portfolio/src/hooks/use-txn-history-dropdowns.tsx

```diff
@@ -0,0 +1,113 @@
+import { useEffect, useState } from 'react';
+import { FormattedMessage } from 'react-intl';
+import { formatTokenType } from '@notional-finance/helpers';
+import { TXN_HISTORY_TYPE } from '@notional-finance/shared-config';
+import { TokenIcon } from '@notional-finance/icons';
+import { useTransactionHistory } from '@notional-finance/notionable-hooks';
+
+export const useTxnHistoryDropdowns = (txnHistoryType: TXN_HISTORY_TYPE) => {
+  const accountHistory = useTransactionHistory();
+  const [currencyOptions, setCurrencyOptions] = useState([]);
+  const [assetOrVaultOptions, setAssetOrVaultOptions] = useState([]);
+
+  let currencyData: any[] = [];
+  let assetData: any[] = [];
+
+  let currencyVaultData: any[] = [];
+  let vaultNameData: any[] = [];
+
+  useEffect(() => {
```

Comment:
I'm not sure this `useEffect` or `useState` calls are necessary here. You should use the `unique` function from `array-helpers.ts` to create a new unique set from the account history objects without needing to do any of this other logic.